bleach/sanitizer.py

```diff
@@ -143,6 +147,25 @@ def __init__(
             alphabetical_attributes=False,
         )
 
+        if css_sanitizer is None:
+            # FIXME(willkg): this doesn't handle when attributes or an
+            # attributes value is a callable
+            attributes_values = []
+            if isinstance(attributes, list):
+                attributes_values = attributes
+
+            elif isinstance(attributes, dict):
+                attributes_values = []
+                for values in attributes.values():
+                    if isinstance(values, (list, tuple)):
+                        attributes_values.extend(values)
+
+            if "style" in attributes_values:
+                warnings.warn(
+                    "'style' attribute specified, but css_sanitizer not set.",
+                    category=NoCssSanitizerWarning,
+                )
```

Comment:
Mmm... I should probably do this in the sanitizer. I'll move it.

Reply:
On second thought, keeping it here means that it only runs once per cleaner and not once per clean, so I'll keep it here.